### ISSUE TITLE
Dashboard Email Reports  Selector displays ID instead of Name

### ISSUE THREAD

**Issue by jawabuu** [apache/superset#13124]
## Screenshot

![image](https://user-images.githubusercontent.com/49921621/107914239-0b832b80-6f73-11eb-947a-4d3befb06165.png)

## Description

When creating a Dashboard Email report the selector displays the ID instead of the name.
For Example `Dashboard <1>` instead of `World Bank's Data`
I have observed this in version `1.0.0` and `1.0.1`
Chart Email reports work just fine.

## Design input
[describe any input/collaboration you'd like from designers, and
tag accordingly. For design review, add the
label `design:review`. If this includes a design proposal,
include the label `design:suggest`]
Display Dashboard Name.
`bug:cosmetic`

**Comment by ktmud** [comment]
There is an open PR for this: https://github.com/apache/superset/pull/11776

Although you can now also try out the new Alert & Report manager by turning on the [`ALERT_REPORTS`](https://github.com/apache/superset/blob/50fa10054fd5d2c28553c4ffd754e12bdedef5da/RELEASING/release-notes-1-0/README.md#feature-flags) feature flag.

**Comment by jawabuu** [comment]
Hey @ktmud 
Thank you
I have the following set up
```
FEATURE_FLAGS = {
    "KV_STORE": True,
    "SHARE_QUERIES_VIA_KV_STORE": True,
    "ENABLE_TEMPLATE_PROCESSING": True,
    "ENABLE_REACT_CRUD_VIEWS": False,
    "ROW_LEVEL_SECURITY": True,
    "ALERT_REPORTS": True,
    "DASHBOARD_NATIVE_FILTERS": True,
    "THUMBNAILS": True,
  }
```
However, when I visit `Settings>Alerts & Reports` I get a 404

![image](https://user-images.githubusercontent.com/49921621/107926359-a9ccbc80-6f86-11eb-9ed0-1c0a10260b6d.png)

**Comment by ktmud** [comment]
Have you turned off `ENABLE_REACT_CRUD_VIEWS` by any chance?

The new alerts and reports kind of depend on it: https://github.com/apache/superset/blob/52ba82fb8b9f1ce5519f1a13f6417a0ce486e7df/superset/views/alerts.py#L80


@lilykuang @nytai do you think whether it makes sense to force turning on `ENABLE_REACT_CRUD_VIEWS` when  `ALERT_REPORTS` is enabled?

**Comment by jawabuu** [comment]
@ktmud I had turned it off. Let me turn it on and retry.

**Comment by jawabuu** [comment]
@ktmud 
I am now able to view the reports and have set up a test report to be sent every minute.
I have celery workers and beat running as well as cache configured.
![image](https://user-images.githubusercontent.com/49921621/107951380-4d2ec900-6fa9-11eb-927a-33555244958b.png)
I have not received any so far, is there any further configuration I need to do?

**Comment by jawabuu** [comment]
Managed to find these from github search

```
from celery.schedules import crontab
CELERYBEAT_SCHEDULE = {
          'email_reports.schedule_hourly': {
              'task': 'email_reports.schedule_hourly',
              'schedule': crontab(minute='*/59'),
          },
          'alerts.schedule_check': {
            'task': 'alerts.schedule_check',
            'schedule': crontab(minute='*/4'')
          },
          'reports.scheduler': {
            'task': 'reports.scheduler',
            'schedule': crontab(minute='*/5'),
          },
          'reports.prune_log': {
              'task': 'reports.prune_log',
              'schedule': crontab(minute=0, hour=0),
          },
      }
```

**Comment by nytai** [comment]
@ktmud We could throw a warning if `ALERT_REPORTS` is enabled, but `ENABLE_REACT_CRUD_VIEWS` is not. I think that fully deprecating the `ENABLE_REACT_CRUD_VIEWS` feature flag makes more sense, I don't think there's any strong case for continuing to support the old FAB views at this point.

**Comment by dpgaspar** [comment]
@nytai agree, planning on removing the MVC CRUD `ModelView` code also. But we can deprecate the flag now

**Comment by zuzana-vej** [comment]
@jawabuu  can you please confirm if this issue is still existing / close this GH# if it's resolved? Thanks

**Comment by jawabuu** [comment]
@zuzana-vej The issue still exists. The PR to close this has not been merged.
